File: LC/Python/Medium/647.py
class Solution:
    def countSubstrings(self, s: str) -> int:
        def expand(start, left, right):
            pal = 0
            while (left >= 0 and right < len(s) and s[left] == s[right]):
                left -= 1
                right += 1
                pal += 1
            return pal

        ans = 0
        
        for i in range(len(s)):
            ans += expand(i, i, i)
            ans += expand(i, i, i+1)
                
        return ans

File: LC/C++/Medium/647.cpp
class Solution {
    int n_;
    std::string s_;
public:
    int expand(int l, int r) {
        int count = 0;

        while (l >= 0 && r < n_ && s_[l] == s_[r]) {
            l--;
            r++;
            count++;
        }

        return count;
    }

    int countSubstrings(string s) {
        n_ = s.size();
        s_ = s;

        int ans = 0;

        for (int i = 0; i < n_; ++i) {
            ans += expand(i, i) + expand(i, i + 1);
        }

        return ans;
    }
};